File: Problem3.java
/*
 * The prime factors of 13195 are 5, 7, 13 and 29.
 * What is the largest prime factor of the number 600851475143 ?
 */

import java.math.BigInteger;

public class Problem3 {

  public static boolean checkPrime (BigInteger p, BigInteger increment) {
  
    BigInteger i = new BigInteger("2");

    while (i.compareTo(p) < 0) {
    
      if ((p.mod(i)).equals(0)) {
      
        return false;
      }

      i = i.add(increment);
    }
    
    return true;
  }

  public static void main(String[] args) {
  
    BigInteger number = new BigInteger("600851475143");
    BigInteger biggestPrime = new BigInteger ("-1");

    BigInteger i = new BigInteger("2");
    BigInteger increment = new BigInteger("1");
    
    while (i.compareTo(number) < 0) {
    
      if ((number.mod(i)).equals(0) && checkPrime(i, increment) == true) { 
      
        if (biggestPrime.compareTo(i) == -1) {
        
          biggestPrime = i;
        }
      }

      i = i.add(increment);
    }
    
    System.out.print(biggestPrime.toString());
  }
}

File: Problem3.py
# The prime factors of 13195 are 5, 7, 13 and 29.
# What is the largest prime factor of the number 600851475143 ?

def Problem3(n) :
    q = next (x for x in range(2, n+1) if (n % x ==0))
    if (n == q) :
        return n
    else:
        return max(largest_p(q),largest_p(int(n/q)))

print(largest_p(600851475143))
